app/celery/provider_tasks.py

```diff
@@ -242,12 +243,16 @@ def _handle_delivery_failure(
         celery_task.request.chain = None
 
     elif isinstance(e, (NullValueForNonConditionalPlaceholderException, AttributeError, RuntimeError)):
-        log_and_update_critical_failure(
-            notification_id,
-            method_name,
-            e,
-            STATUS_REASON_UNDELIVERABLE,
-        )
+        if 'Duplication prevention' in str(e):
```

Comment:
Should this be a constant

Reply:
Had the same thought when making it, decided to keep it simple as this was a very specific error that was raised for only this use case.